### PR TITLE
Remove gfx funcs. Use `tempContext` more

### PR DESCRIPTION
**Pull request by TimAbraldes**
Remove `withFont` and `measureWidth`

Use `tempContext` to canonicalize font CSS strings.

Use `tempContext` when measuring text.

This PR doesn't have much impact on its own but it will make a future change easier.

Startup benchmark 35 rounds 2500ms:

|      Test     | Baseline Mean |   Mean   |  +/-  |   %   |       P       |    Min   |    Max   | 
|:--------------|--------------:|---------:|------:|------:|--------------:|---------:|---------:|
| startupTime   |       1,823ms |  1,958ms | 135ms |  7.39 | INSIGNIFICANT |  1,611ms |  2,506ms | 
| totalSize     |      29,276kb | 29,305kb |  29kb |  0.10 | INSIGNIFICANT | 27,728kb | 30,520kb | 
| domSize       |         123kb |    123kb |   0kb |  0.01 | INSIGNIFICANT |    122kb |    123kb | 
| styleSize     |         435kb |    444kb |   9kb |  2.07 |         WORSE |    443kb |    450kb | 
| jsObjectsSize |      17,515kb | 17,520kb |   4kb |  0.03 |         WORSE | 17,505kb | 17,560kb | 
| jsStringsSize |         735kb |    731kb |  -4kb | -0.55 |        BETTER |    730kb |    734kb | 
| jsOtherSize   |      10,105kb | 10,107kb |   3kb |  0.03 | INSIGNIFICANT |  8,547kb | 11,308kb | 
| otherSize     |         364kb |    381kb |  17kb |  4.58 |         WORSE |    378kb |    448kb |

### CODE REVIEW

**Comment by marco-c** [comment]
Unfortunately the result is "INSIGNIFICANT", this means that there's no way to tell whether it's better or worse.
Since the mean is higher with the change, the results could be actually worse.
Could you run the benchmark on device as well?

**Comment by TimAbraldes** [comment]
Startup bench on a complex MIDlet, 100 rounds, 10s round delay:

|      Test     | Baseline Mean |   Mean   |  +/-  |   %   |       P       |    Min   |    Max   | 
|:--------------|--------------:|---------:|------:|------:|--------------:|---------:|---------:|
| startupTime   |       1,641ms |  1,645ms |   4ms |  0.22 | INSIGNIFICANT |  1,510ms |  1,983ms | 
| totalSize     |      28,894kb | 28,908kb |  15kb |  0.05 | INSIGNIFICANT | 27,314kb | 29,552kb | 
| domSize       |         122kb |    122kb |   0kb | -0.00 | INSIGNIFICANT |    122kb |    123kb | 
| styleSize     |         435kb |    443kb |   9kb |  1.99 |         WORSE |    443kb |    443kb | 
| jsObjectsSize |      17,371kb | 17,379kb |   8kb |  0.05 |         WORSE | 17,362kb | 17,426kb | 
| jsStringsSize |         666kb |    659kb |  -7kb | -0.99 |        BETTER |    657kb |    669kb | 
| jsOtherSize   |       9,936kb |  9,925kb | -11kb | -0.11 | INSIGNIFICANT |  8,348kb | 10,521kb | 
| otherSize     |         363kb |    379kb |  15kb |  4.24 |         WORSE |    377kb |    395kb |

File: midp/gfx.js [hunk 1]
```diff
@@ -405,8 +407,8 @@ var currentlyFocusedTextEditor;
     };
 
     Native["javax/microedition/lcdui/Font.charWidth.(C)I"] = function(char) {
-        withFont(this, MIDP.context2D);
-        return measureWidth(MIDP.context2D, String.fromCharCode(char));
+        tempContext.font = this.css;
```
Comment (marco-c) [inline midp/gfx.js]: withFont was optimized by avoiding to set the *font* property if not necessary. Setting it was somehow slower (IIRC @yxl implemented this optimization)

File: midp/gfx.js [hunk 2]
```diff
@@ -441,28 +443,18 @@ var currentlyFocusedTextEditor;
...
     function withTextAnchor(g, c, anchor, x, y, str) {
-        withFont(g.currentFont, c);
+        c.font = g.currentFont.css;
 
         c.textAlign = "left";
         c.textBaseline = "top";
 
         if (anchor & RIGHT || anchor & HCENTER) {
-            var w = measureWidth(c, str);
+            var w = c.measureText(str).width;
```
Comment (marco-c) [inline midp/gfx.js]: The width used to be truncated with "| 0". Integer coordinates used to be faster than floating point coordinates on mobile, I'm not sure if this is still true.

**Comment by marco-c** [comment]
The result is still "INSIGNIFICANT", but now the means are pretty similar so a regression seems unlikely.

File: midp/gfx.js (hunk 1 above)
Comment (yxl) [inline midp/gfx.js]: It is quit simple to fix:
```JavaScript
if (tempContext.font != this.css) {
  tempContext.font = this.css;
}
```
see https://github.com/mozilla/j2me.js/pull/721

**Comment by TimAbraldes** [comment]
Closing in favor of #1395 - that PR contains these changes plus the changes that this patch enables

**Pull request closed without merging**